enumerables.rb

```diff
@@ -0,0 +1,101 @@
+module Enumerable
+
+    def my_each
+        for i in 0..self.length-1
+            yield(self[i])
+        end
+    end
+    
+    def my_each_with_index
+        for i in 0..self.length-1
+            yield(self[i],i)
+        end
+    end
+
+    def my_select
+        arr = []
+        for i in 1..self.length
+            if yield(i)
+                arr.push(i)
+            end
+        end
+        arr
+    end
+
+    def my_all?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = true
+            end
+        end
+        results
+    end
+
+    def my_any?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = true
+            end
+        end
+        results
+    end
+
+    def my_none?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = false
+            end
+        end
+        results
+    end
+
+    def my_count
+        counter = 0
+        for i in 0..self.length-1
+            if yield(self[i])
+                counter +=1
+            end
+        end
+        counter
+    end
+
+    def my_map(proc= nil)
+        array = []
+        for i in 1..self.length
+            yield(self[i])
+            array.push(i)
+        end
+        array
+    end
```

Comment:
* [ ] my_map returns an Enumerator if no block is given
   ```ruby
    array = [1,5,8,7,9,5,5,55,5]
    p array.my_map #error
    LocalJumpError:
        no block given (yield)
  ```

* [ ] my_map returns a new array with the results of running block once for every element in enum.
  ```ruby
    array = [1,2,5,4,7]
    block = proc { |num| num < 5 }
    p array.my_map?(&block) # returns this error
       #LocalJumpError:
          #no block given (yield)
      # ./enumerables.rb:68:in `block in my_map'
      # ./enumerables.rb:67:in `each'
      # ./enumerables.rb:67:in `my_map'
  ```